### PR TITLE
feat: cherry pick changes from master to support kubernetes 1.18

### PR DESCRIPTION
**Pull request by bowen5**
**Reason for Change**:

Cherry pick changes from aks-engine master branch to bring the kubernetes 1.18 support into this branch.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by bowen5** [comment]
@xuto2, I didn't pick the csi-proxy related PR https://github.com/Azure/aks-engine/pull/2930/files. Please let me know if I need to do so. Thank you.

File: parts/k8s/addons/1.18/kubernetesmasteraddons-kube-proxy-daemonset.yaml [hunk 1]
```diff
@@ -9,6 +9,15 @@ data:
...
     clusterCIDR: "{{ContainerConfig "cluster-cidr"}}"
     mode: "{{ContainerConfig "proxy-mode"}}"
+    {{- if ContainerConfig "bind-address"}}
+    bindAddress: "{{ContainerConfig "bind-address"}}"
+    {{end}}
+    {{- if ContainerConfig "healthz-bind-address"}}
+    healthzBindAddress: "{{ContainerConfig "healthz-bind-address"}}"
+    {{end}}
+    {{- if ContainerConfig "metrics-bind-address"}}
+    metricsBindAddress: "{{ContainerConfig "metrics-bind-address"}}"
```
Comment (xuto2) [inline parts/k8s/addons/1.18/kubernetesmasteraddons-kube-proxy-daemonset.yaml]: Just FYI - this change, if deemed need by 1.18, should be ported to AKS RP CCP charts.

File: vhd/packer/install-dependencies.sh [hunk 2]
```diff
@@ -224,6 +224,7 @@ for TILLER_VERSION in ${TILLER_VERSIONS}; do
 done
 
 CLUSTER_AUTOSCALER_VERSIONS="
+1.18.0
```
Comment (xuto2) [inline vhd/packer/install-dependencies.sh]: changes in this file should go to https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/packer/install-dependencies.sh as AKS build its own VHD

**Comment by xuto2** [comment]
> @xuto2, I didn't pick the csi-proxy related PR https://github.com/Azure/aks-engine/pull/2930/files. Please let me know if I need to do so. Thank you.

I think you can cherry pick it, as it's considered part of 1.18 feature.

**Comment by xuto2** [comment]
as we talked offline we don't want non-urgent changes to be merged to the branch aks-release-v0.47.0-1 branch during the code freeze, so I create a new branch https://github.com/Azure/aks-engine/tree/aks-release-v0.47.1 for you to hold your change and you can vendor this specific branch to aks-rp for your private changes in RP.  Once the code freeze is over, we can backfill it and vendor it to aks RP. Let me know if you have any concerns.

File: parts/k8s/addons/1.18/kubernetesmasteraddons-kube-proxy-daemonset.yaml (hunk 1 above)
Comment (bowen5) [reply parts/k8s/addons/1.18/kubernetesmasteraddons-kube-proxy-daemonset.yaml]: Thank you for the info.

**Comment by bowen5** [comment]
> as we talked offline we don't want non-urgent changes to be merged to the branch aks-release-v0.47.0-1 branch during the code freeze, so I create a new branch https://github.com/Azure/aks-engine/tree/aks-release-v0.47.1 for you to hold your change and you can vendor this specific branch to aks-rp for your private changes in RP. Once the code freeze is over, we can backfill it and vendor it to aks RP. Let me know if you have any concerns.

Thank you Tony! Closing the PR as discussed and creating a new PR to v0.47.1

**Comment by bowen5** [comment]
> > @xuto2, I didn't pick the csi-proxy related PR https://github.com/Azure/aks-engine/pull/2930/files. Please let me know if I need to do so. Thank you.
> 
> I think you can cherry pick it, as it's considered part of 1.18 feature.

Sure.

**Pull request closed without merging**